api/dps/index_test.go

```diff
@@ -131,413 +123,387 @@ func TestIndex_Last(t *testing.T) {
 
+		codec := mocks.BaselineCodec(t)
+		codec.UnmarshalFunc = cbor.Unmarshal
+
 		index := Index{
+			codec: codec,
 			client: &apiMock{
 				GetHeaderFunc: func(_ context.Context, in *GetHeaderRequest, _ ...grpc.CallOption) (*GetHeaderResponse, error) {
-					assert.Equal(t, testHeader.Height, in.Height)
+					assert.Equal(t, mocks.GenericHeader.Height, in.Height)
 
 					return &GetHeaderResponse{
-						Height: testHeader.Height,
-						Data:   testHeaderB,
+						Height: mocks.GenericHeader.Height,
+						Data:   headerB,
 					}, nil
 				},
 			},
-			codec: &mocks.Codec{
-				UnmarshalFunc: cbor.Unmarshal,
-			},
 		}
 
-		got, err := index.Header(testHeader.Height)
+		got, err := index.Header(mocks.GenericHeader.Height)
 
 		if assert.NoError(t, err) {
-			assert.Equal(t, testHeader, got)
+			assert.Equal(t, mocks.GenericHeader, got)
 		}
 	})
 
 	t.Run("handles index failures", func(t *testing.T) {
 		t.Parallel()
 
 		index := Index{
+			codec: mocks.BaselineCodec(t),
 			client: &apiMock{
 				GetHeaderFunc: func(_ context.Context, in *GetHeaderRequest, _ ...grpc.CallOption) (*GetHeaderResponse, error) {
-					assert.Equal(t, testHeader.Height, in.Height)
+					assert.Equal(t, mocks.GenericHeader.Height, in.Height)
 
-					return nil, mocks.DummyError
+					return nil, mocks.GenericError
 				},
 			},
 		}
 
-		_, err := index.Header(testHeader.Height)
+		_, err := index.Header(mocks.GenericHeader.Height)
 
 		assert.Error(t, err)
 	})
 
 	t.Run("handles decoding failures", func(t *testing.T) {
 		t.Parallel()
 
+		codec := mocks.BaselineCodec(t)
+		codec.UnmarshalFunc = cbor.Unmarshal
+
 		index := Index{
+			codec: codec,
 			client: &apiMock{
 				GetHeaderFunc: func(_ context.Context, in *GetHeaderRequest, _ ...grpc.CallOption) (*GetHeaderResponse, error) {
-					assert.Equal(t, testHeader.Height, in.Height)
+					assert.Equal(t, mocks.GenericHeader.Height, in.Height)
 
 					return &GetHeaderResponse{
-						Height: testHeader.Height,
+						Height: mocks.GenericHeader.Height,
 						Data:   []byte(`invalid data`),
 					}, nil
 				},
 			},
-			codec: &mocks.Codec{
-				UnmarshalFunc: cbor.Unmarshal,
-			},
 		}
 
-		_, err := index.Header(testHeader.Height)
+		_, err := index.Header(mocks.GenericHeader.Height)
 
 		assert.Error(t, err)
 	})
 }
 
 func TestIndex_Commit(t *testing.T) {
-	testHeight := uint64(42)
-	testCommit, err := flow.ToStateCommitment([]byte("07018030187ecf04945f35f1e33a89dc"))
-	require.NoError(t, err)
-
 	t.Run("nominal case", func(t *testing.T) {
 		t.Parallel()
 
+		commit := mocks.GenericCommit(0)
 		index := Index{
 			client: &apiMock{
 				GetCommitFunc: func(_ context.Context, in *GetCommitRequest, _ ...grpc.CallOption) (*GetCommitResponse, error) {
-					assert.Equal(t, testHeight, in.Height)
+					assert.Equal(t, mocks.GenericHeight, in.Height)
 
 					return &GetCommitResponse{
-						Height: testHeight,
-						Commit: testCommit[:],
+						Height: mocks.GenericHeight,
+						Commit: commit[:],
```

Comment:
I think it depends on the Flow version whether this is needed. Anyway, maybe we could add a function for this, something like:

```go
func ByteSlice(v interface{}) []byte {
  switch vv := v.(type) {
    case flow.Identifier:
      return vv[:]
    case flow.StateCommitment:
      return vv [:]
    // ....
    default:
     panic("invalid type")
  }
}
```

WDYT?

Reply:
Ah, yeah I like that idea, I hadn't thought of that :)